src/vcpkg/vcpkgcmdarguments.cpp

```diff
@@ -854,19 +848,25 @@ namespace vcpkg
         else
         {
             Json::Object obj;
-            if (args.vcpkg_root_dir)
+            if (auto vcpkg_root_dir_arg = args.vcpkg_root_dir_arg.get())
+            {
+                obj.insert(VCPKG_ROOT_ARG_NAME, Json::Value::string(*vcpkg_root_dir_arg));
+            }
+
+            if (auto vcpkg_root_dir_env = args.vcpkg_root_dir_env.get())
             {
-                obj.insert(VCPKG_ROOT_DIR_ENV, Json::Value::string(*args.vcpkg_root_dir.get()));
+                obj.insert(VCPKG_ROOT_ENV_NAME, Json::Value::string(*vcpkg_root_dir_env));
             }
```

Comment:
These should probably be consolidated.

Reply:
The problem is that the decision of what to use depends on information that isn't known until later, in `VcpkgPaths`.

Reply:
(To clarify: Yes, the right thing to do is to determine what the correct thing is, and set that. I went with this more hacky solution so that I could get this out quickly without needing to substantially alter the relationship between VcpkgCmdArguments and VcpkgPaths.